### PR TITLE
Card Loading API

### PR DESCRIPTION
**Pull request by RaitoBezarius**
Fixes partially #130 (a plan is explained over there).

I think we should get rid of JSReverse in another PR which will clean out the JavaScript to use the public API.

### CODE REVIEW

File: mangaki/mangaki/settings.py [hunk 1]
```diff
@@ -149,13 +150,22 @@
 LOGIN_URL = '/user/login/'
 LOGIN_REDIRECT_URL = '/'
 ACCOUNT_EMAIL_REQUIRED = True
-ACCOUNT_SIGNUP_FORM_CLASS= 'mangaki.forms.SignupForm'
+ACCOUNT_SIGNUP_FORM_CLASS = 'mangaki.forms.SignupForm'
```
Comment (RaitoBezarius) [inline mangaki/mangaki/settings.py]: sorry, my OCD is incurable.

File: mangaki/mangaki/api/cards.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+from mangaki.models import Category, Work
+from mangaki.utils.ratings import current_user_ratings
+
+NB_POINTS_DPP = 10
+POSTERS_PER_PAGE = 24
+
+
+class SlotCardTypes(enum.Enum):
+    popularity = 'popular'
+    controversy = 'controversial'
```
Comment (Elarnon) [inline mangaki/mangaki/api/cards.py]: Why do we have different values than the name?

File: mangaki/mangaki/static/js/vote.js [hunk 3]
```diff
@@ -210,6 +210,10 @@ Slot.prototype.fetch = function () {
 /* Mosaic takes care of mapping Cards inside an element with Slots pointing to
  * some remote URLs on the server.
  */
+function buildSlotURL(category, slot_sort) {
```
Comment (Elarnon) [inline mangaki/mangaki/static/js/vote.js]: Can you keep the documentation for the `Mosaic` function close to the `Mosaic` function?

File: mangaki/mangaki/views.py [hunk 4]
```diff
@@ -317,8 +272,8 @@ def get_context_data(self, **kwargs):
 
         if sort_mode == 'mosaic' and not self.is_dpp:
             context['object_list'] = [
-                Work(title='Chargement…', ext_poster='/static/img/chiro.gif')
-                for _ in range(4)
+                (slot_sort_type, Work(title='Chargement…', ext_poster='/static/img/chiro.gif'))
```
Comment (Elarnon) [inline mangaki/mangaki/views.py]: Can we use `{'sort_type': slot_sort_type, 'work': Work(title=...)}` instead? Would make the template code more readable. BTW we should do `{'work': Work(...)}` for the other cases as well for consistency; that way we can easily get rid of `works_mosaic.html`.

File: mangaki/mangaki/templates/works_mosaic.html [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<div class="row">
+  <div class="col-xs-12 cards-grid">
+    {% for slot_and_work in works %}
```
Comment (Elarnon) [inline mangaki/mangaki/templates/works_mosaic.html]: If we get rid of the tuple we can just do `{% for work_meta in works %}` in `works.html` then pass the `work=work_meta.work slot_type=work_meta.slot_type` in the `{% include %}` below. I believe Django uses `None` as the default for missing keys.

File: mangaki/mangaki/api/cards.py (hunk 2 above)
Comment (RaitoBezarius) [inline mangaki/mangaki/api/cards.py]: At the start, I wanted to have something like: `queryset[SlotCardTypes.popularity]()`

But, due to DPP and random special treatment, that does not sound possible.

As a relic of the past, I kept the mapping. Shouldn't we uniformize them to match the queryset methods?

File: mangaki/mangaki/static/js/vote.js (hunk 3 above)
Comment (RaitoBezarius) [inline mangaki/mangaki/static/js/vote.js]: Sure.

File: mangaki/mangaki/views.py (hunk 4 above)
Comment (RaitoBezarius) [inline mangaki/mangaki/views.py]: Good idea.

File: mangaki/mangaki/static/js/vote.js [hunk 6]
```diff
@@ -219,8 +223,8 @@ function Mosaic(el, category) {
 
   var els = $(el).find('.work-card').toArray();
 
-  this.slots = els.map(function (el, index) {
-    return new Slot('/data/card/' + category + '/' + (index + 1) + '.json');
+  this.slots = els.map(function (el) {
+    return new Slot(buildSlotURL(category, $(el).data('slot-sort')));
```
Comment (Elarnon) [inline mangaki/mangaki/static/js/vote.js]: Please use attr('data-slot-sort')

Comment (Elarnon) [inline mangaki/mangaki/static/js/vote.js]: el.getAttribute('data-slot-sort')

**Pull request merged**